bin/backup.sh

```diff
@@ -5,6 +5,7 @@ set -e
 setup.sh
 
 max_pg_wait_count=120
+work_area=${PGDUMP_BACKUP_AREA:-/pg_dump}
```

Comment:
Why does the work area need to be configurable? Do you want to bind mount it from a different host volume, and are then unable to `rm -rf` the bind mounted directory?

Maybe we just `rm /pg_dump/*.sql`, instead? And then you can bind mount `/pg_dump` from anywhere, and we can also `mkdir /pg_dump` in `Dockerfile` and no longer need `mkdir -p`.

Reply:
Because we're wanting this in a different volume, and force making this in `/` is a thing that only makes sense inside a container, and a pretty specific container at that. While this code _is_ for a container, making it dependent on that is a pain point waiting to happen (as has just happened to us).

Making the path configurable lets you tune things without hand configuring the container build with a bind mount. Far better to have the script be somewhat generic by being able to tell it where to work.

One stong advantage of the mkdir is that we _know_ we make the work and nobody else might be using it. Otherwise the `rm -rf` below, or your suggested `rm *.sql` are just asking to damage something else's work. With mkdir, we made it and we're free to unmake it.

Reply:
I am not sure I follow. Are you using this script outside of a container? From what I have seen, it is fairly common for Docker images to store data in a subdirectory of `/`.

As far as I can tell, there is no point in making the directory configurable *unless* you are wanting to use a bind mount, which must be hand configured by you in your compose file or `docker run` command, *and* needing to completely remove and recreate that directory which would require you nominate a subdirectory of the mount point.

I think we could just use `/data/pg_dump` as the directory and you could then use `-v /some/host/dir:/data` and we could continue to destroy and recreate the `pg_dump` subdirectory, without this extra configuration or documentation.

Fair point about the safety of retaining the mkdir and exit if it already exists. Though, if something ever goes wrong and a container exits (e.g. OOM killed) before having removed the directory, the container will not be able to restart resume backups without intervention?